migration_engine/adapters/sqlite3_adapter.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+from typing import Literal
+from dataclasses import dataclass, field
+from adapters.base_adapter import BaseAdapter
+
+
+type DataTypes = Literal["TEXT", "INTEGER", "REAL", "BLOB", "NULL"]
+type Constraints = Literal["PRIMARY KEY", "UNIQUE", "NOT NULL", "DEFAULT"]
+
+
+class SQLite3Adapter(BaseAdapter):
+
+    @dataclass
+    class Column:
+        """Column dataclass for SQLite3Adapter"""
+
+        name: str
+        data_type: DataTypes
+        constraints: list[Constraints] = field(default_factory=list)
+
+    @dataclass
+    class Table:
+        """Table dataclass for SQLite3Adapter"""
+
+        name: str
+        columns: list[SQLite3Adapter.Column] = field(default_factory=list)
```

Comment:
This code is repated from BaseAdapter. Maybe try to make a generic Column and Table in Base adapter and here, just specify its type arguments.